File: 643-maximum-average-subarray-i/maximum-average-subarray-i.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        cnt = 0
        maxim = float("-inf")
        average = float("-inf")
        for i in range(len(nums)):
            cnt += nums[i]
            if i >= k:
                cnt -=nums[i-k]
                maxim = max(maxim, cnt)
                average = max(average, maxim/k)
            if i ==k-1:
                maxim = max(maxim, cnt)
                average = max(average, maxim/k)
        return average

File: 643-maximum-average-subarray-i/maximum-average-subarray-i.cpp
class Solution {
public:
    double findMaxAverage(vector<int>& nums, int k) {
        double average;
        double k_sum = 0;
        for (int i=0; i <k; i++){
            k_sum +=nums[i];
        }

        average = double(k_sum)/k;
        for(int i = k; i < nums.size(); i++){
            k_sum +=nums[i]  - nums[i-k];
            average = max(average, double(k_sum)/k);
        }

        return average;

        
    }
};